src/main.rs

```diff
@@ -76,7 +77,69 @@ async fn runWebServer() -> Result<()> {
     Ok(())
 }
 
+// TODO: remove these references :-)
+// See: https://github.com/keylime/keylime/blob/master/keylime/revocation_notifier.py - await_notifications
+//      https://github.com/keylime/rust-keylime/issues/92
+//      https://github.com/erickt/rust-zmq/blob/master/examples/zguide/pathosub/main.rs
 async fn runRevocationService() -> Result<()> {
+
+    let mount = secure_mount::mount().unwrap();
+    let revocation_cert_path = format!("{}/unzipped/RevocationNotifier-cert.crt", mount);
+
+    // TODO: should this be a different type?
+    let mut cert_key = String::from("");
+
+    let context = zmq::Context::new();
+    let mysock = context.socket(zmq::SUB).unwrap();
+    mysock
+        .set_subscribe(b"")
+        .expect("failed setting the 0mq subscription");
+
+    let revocation_ip = config_get("/etc/keylime.conf", "general", "receive_revocation_ip")?;
+    let revocation_port = config_get("/etc/keylime.conf", "general", "receive_revocation_port")?;
+    let endpoint = format!("{}:{}", revocation_ip, revocation_port);
+
+    mysock
+        .connect(endpoint.as_str())
+        .expect("could not connect to revocation service");
+
+    info!("Waiting for revocation messages on 0mq {}", endpoint);
+
+    loop {
+        let rawbody = mysock
+            .recv_string(0)
+            .expect("failed to receive 0mq message")
+            .unwrap();
+        let body: Value = serde_json::from_str(rawbody.as_str())?;
+        if cert_key == "" {
+            // load up the CV signing public key
+            if revocation_cert_path != "" && Path::new(&revocation_cert_path).exists() {
```

Comment:
`revocation_cert_path` can't be empty -- will drop the former and keep the later check